tutorial1.md

```diff
@@ -0,0 +1,299 @@
+
+A container isn't enough to run this, so we'll need a pod spec as well.
+The `pod` is something you may find yourself writing, and there's even an [API page][podapi]!
+
+[simpleservice]: https://hub.docker.com/r/mhausenblas/simpleservice/
+[podapi]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#pod-v1-core
+
+You should end up with something like this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: myapp-pod
+  labels:
+    app: myapp
+spec:
+  containers:
+  - name: myapp-container
+    image: mhausenblas/simpleservice:0.5.0
+    ports:
+    - containerPort: 8080
+    env:
+    - name: PORT0
+      value: "8080"
+```
+
+Kubelet will look for that file and start the pod. You can see this in the output from docker:
+
+```console
+$ sudo docker ps
+CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS               NAMES
+f96c7c333214        mhausenblas/simpleservice   "python ./simpleserv…"   7 minutes ago       Up 7 minutes                            k8s_myapp-container_myapp-pod-k8s-tutorial_default_7a32a55f16cc86f3976f4b8e2ee88408_0
+586a9baa3e2e        k8s.gcr.io/pause:3.1        "/pause"                 7 minutes ago       Up 7 minutes                            k8s_POD_myapp-pod-k8s-tutorial_default_7a32a55f16cc86f3976f4b8e2ee88408_0
+```
+
+NOTE: Probably most of the hashes here will be different. But the important this is the `k8s_myapp-container_myapp-pod`, which means our pod is running!
+
+You'll notice the ports aren't exposed. Never fear, we can still access our service! First, get a shell running on the docker container:
+
+```console
+$ sudo docker exec -it f96c7c333214 bash
+#
+```
+
+NOTE: the `#` means that this is a root shell. Your prompt will be longer, probably somthing like `root@myapp-pod-k8s-tutorial:/usr/src/app#`. This has been omitted for brevity.
+
+next, retrieve the IP address:
+
+```console
+# ip addr
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+7: eth0@if8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
+    link/ether 02:42:ac:11:00:02 brd ff:ff:ff:ff:ff:ff
+    inet 172.17.0.2/16 brd 172.17.255.255 scope global eth0
+       valid_lft forever preferred_lft forever
+```
+
+NOTE: Your IP address may be different, but it should still be in the `172.0.0.0/8` subnet.
+
+Now, log out of that machine:
+
+```console
+# exit
+$
+```
+
+And you should be able to curl our endpoint!
+
+```console
+$ curl -i http://172.17.0.2:8080/endpoint0
+HTTP/1.1 200 OK
+Date: Sat, 23 Mar 2019 03:32:56 GMT
+Content-Length: 72
+Etag: "246ff893b8e03857ceb41ac80403c24d5e255452"
+Content-Type: application/json
+Server: TornadoServer/4.3
+
+{"host": "172.17.0.2:8080", "version": "0.5.0", "result": "all is well"}
+```
```

Comment:
some text now to summarize and say that we've gotten a kubelet to start up, and receive instructions on static pods to run, which has taught us about how a kubelet is configured and how to make it slightly useful!